src/python/pants/backend/python/rules/targets.py

```diff
@@ -0,0 +1,207 @@
+    `__main__` function.
+    """
+
+    alias: ClassVar = "entry_point"
+
+
+class Platforms(StringOrStringListField):
+    """Extra platforms to target when building a Python binary."""
+
+    alias: ClassVar = "platforms"
+
+
+class PexInheritPath(BoolField):
+    """Whether to inherit the `sys.path` of the environment that the binary runs in or not."""
+
+    alias: ClassVar = "inherit_path"
+    default: ClassVar = False
+
+
+class PexZipSafe(BoolField):
+    """Whether or not this binary is safe to run in compacted (zip-file) form."""
+
+    alias: ClassVar = "zip_safe"
+    default: ClassVar = True
+
+
+class PexAlwaysWriteCache(BoolField):
+    """Whether Pex should always write the .deps cache of the Pex file to disk or not."""
+
+    alias: ClassVar = "always_write_cache"
+    default: ClassVar = False
+
+
+class PexRepositories(StringOrStringListField):
+    """Repositories for Pex to query for dependencies."""
+
+    alias: ClassVar = "repositories"
+
+
+class PexIndices(StringOrStringListField):
+    """Indices for Pex to use for packages."""
+
+    alias: ClassVar = "indices"
+
+
+class IgnorePexErrors(BoolField):
+    """Should we ignore when Pex cannot resolve dependencies?"""
+
+    alias: ClassVar = "ignore_errors"
+    default: ClassVar = False
+
+
+class PexShebang(StringField):
+    """For the generated Pex, use this shebang."""
+
+    alias: ClassVar = "shebang"
+
+
+# TODO: This option is weird. Its default is determined by `--python-binary-pex-emit-warnings`.
+#  How would that work with the Target API? Likely, make this an AsyncField and in the rule
+#  request the corresponding subsystem. For now, we ignore the option.
+class EmitPexWarnings(BoolField):
+    """Whether or not to emit Pex warnings at runtime."""
+
+    alias: ClassVar = "emit_warnings"
+    default: ClassVar = True
+
+
+COMMON_PYTHON_FIELDS = (*COMMON_TARGET_FIELDS, Compatibility, Provides)
+
+
+class PythonBinary(Target):
+    """A Python target that can be converted into an executable Pex file.
+
+    Pex files are self-contained executable files that contain a complete Python
+    environment capable of running the target. For more information about Pex files, see
+    http://pantsbuild.github.io/python-readme.html#how-pex-files-work.
+    """
+
+    alias: ClassVar = "python_binary"
+    core_fields: ClassVar = (
+        *COMMON_PYTHON_FIELDS,
+        PythonBinarySources,
+        EntryPoint,
+        Platforms,
+        PexInheritPath,
+        PexZipSafe,
+        PexAlwaysWriteCache,
+        PexRepositories,
+        PexIndices,
+        IgnorePexErrors,
+        PexShebang,
+        EmitPexWarnings,
+    )
+
+
+class PythonLibrary(Target):
+    alias: ClassVar = "python_library"
+    core_fields: ClassVar = (*COMMON_PYTHON_FIELDS, PythonLibrarySources)
+
+
+class PythonTests(Target):
+    alias: ClassVar = "python_tests"
+    core_fields: ClassVar = (*COMMON_PYTHON_FIELDS, PythonTestsSources, Coverage, Timeout)
+
+
+def targets():
```

Comment:
Where is this called? I would have expected it in a `register.py`?

Reply:
Nowhere, yet. It will soon indeed be in `register.py`.